src/sample_weight_audit/estimator_check.py

```diff
@@ -172,12 +179,37 @@ def compute_predictions(est, X):
             return est.predict_proba(X)
         else:
             return est.decision_function(X)
+    elif isinstance(est, Pipeline):
+        if score:
+            return mean_squared_error(est.predict(X), y)
```

Comment:
This is only valid for regressors and should therefore go into the regressor branch, irrespective of whether the estimator is a pipeline or not.

For classifiers, we should leverage maximally informative, threshold independent classification metrics such as `log_loss` on the output of `predict_proba` and `roc_auc_score` on the output of `decision_function`.

Reply:
And for clusterers, we should use clustering metrics that are invariant to label permutations (e.g. Rand index, NMI, V-measure and the like).